### PR TITLE
Feat(eos_cli_config_gen): Add support for monitor layer1

### LINKED ISSUE #3338: Add support for SYSLOG based alerting on optical thresholds
### Enhancement summary

At the moment there is no built-in support in `eos_cli_config_gen` for the SYSLOG based alerting on optical thresholds which is documented [here](https://www.arista.com/en/support/toi/eos-4-21-5f/14180-syslog-based-alerting-on-optical-thresholds)

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

For deployments where SYSLOG messages are required for optical thresholds the only option right now is to use `eos_cli`. The ability to set this natively in AVD rather than using raw `eos_cli` would make it easier to maintain and also validate user input.

### Describe the solution you would like

Ideally a new schema fragment and templates would be added to support this configuration.

Example:
```
monitor_layer1:
       mac_fault: <boolean>
      transciever: <boolean>
```

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

Adding monitor_layer1 data-model

## Related Issue(s)

Fixes #3338 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## How to test

molecule converge -s eos_cli_config_gen

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/monitor-layer1.j2 [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{# doc - monitor layer1 #}
+{% if monitor_layer1.enabled is arista.avd.defined(true) %}
+
+## Monitor Layer 1 Logging
+
+| Layer 1 Event | Logging |
+| ------------- | ------- |
+| mac fault | {{ monitor_layer1.logging_mac_fault | arista.avd.default(false) }} |
+| transceiver dom | {{ monitor_layer1.logging_transceiver.dom | arista.avd.default(false) }} |
+| transceiver communication | {{ monitor_layer1.logging_transceiver.communication | arista.avd.default(false) }} |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/monitor-layer1.j2]:
```suggestion
| MAC fault | {{ monitor_layer1.logging_mac_fault | arista.avd.default(false) }} |
| Transceiver DOM | {{ monitor_layer1.logging_transceiver.dom | arista.avd.default(false) }} |
| Transceiver communication | {{ monitor_layer1.logging_transceiver.communication | arista.avd.default(false) }} |
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/monitor-layer1.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+   ip address 10.73.255.122/24
+```
+
+## Monitor Layer 1 Logging
+
+| Layer 1 Event | Logging |
+| ------------- | ------- |
+| mac fault | True |
+| transceiver dom | True |
+| transceiver communication | True |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/monitor-layer1.md]:
```suggestion
| MAC fault | True |
| Transceiver DOM | True |
| Transceiver communication | True |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/monitor-layer1.md
Comment (emilarista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/monitor-layer1.md:1]: I think this table needs to get inserted into the docs/input-variables.md file so it actually gets featured in the generated documentation on avd.arista.com and the CI-generated docs site. Apart from this, LGTM!

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/monitor-layer1.md:1]: Good catch. @Shivani-gslab please make sure to add the table to the input-variables.md file. I think there is a `monitoring` section or similar.

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/monitor-layer1.md:1]: Added, please check the sequence once.

**Review by emilarista — APPROVED** [review]
LGTM!

File: ansible_collections/arista/avd/roles/eos_designs/schemas/eos_designs.schema.pickle
Comment (chetryan) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/eos_designs.schema.pickle:1]: I think the pickles should no longer be added to PRs.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.pickle
Comment (chetryan) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.pickle:1]: I don't think the pickle should be included in a PR

**Review by chetryan — CHANGES_REQUESTED** [review]
We should remove the pickle file. Otherwise looks good.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/eos_designs.schema.pickle
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/eos_designs.schema.pickle:1]: Removed it. thanks

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.pickle
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.pickle:1]: Removed!

**Review by chetryan — APPROVED** [review]
LGTM!

**Pull request merged**